tests/unit/templating/test_resource_context_builder.py

```diff
@@ -0,0 +1,251 @@
+        builder.add_system_property('sysC', 'C')
+        self.assertEqual(builder.result, {
+            'propA': 'A Prop',
+            'systemProperties': {
+                'sysA': 'A',
+                'sysB': 'B',
+                'sysC': 'C'
+            }, 
+            'deploymentLocationInst': {
+                'name': 'Test',
+                'type': 'Kubernetes',
+                'properties': {
+                    'dlPropA': 'A DL Prop'
+                }
+            }
+        })
+
+    def test_set_deployment_location(self):
+        system_properties, properties, deployment_location = self.__example_values()
+        builder = ResourceContextBuilder(system_properties, properties, deployment_location)
+        new_deployment_location = {
+            'name': 'Alternative',
+            'type': 'Docker',
+            'properties': {
+                'dlPropA': 'Alternative DL Prop'
+            }
+        }
+        builder.set_deployment_location(new_deployment_location)
+        self.assertEqual(builder.result, {
+            'propA': 'A Prop',
+            'systemProperties': {
+                'sysA': 'A',
+                'sysB': 'B'
+            }, 
+            'deploymentLocationInst': {
+                'name': 'Alternative',
+                'type': 'Docker',
+                'properties': {
+                    'dlPropA': 'Alternative DL Prop'
+                }
+            }
+        })
+
+    def test_add_deployment_location_property(self):
+        system_properties, properties, deployment_location = self.__example_values()
+        builder = ResourceContextBuilder(system_properties, properties, deployment_location)
+        builder.add_deployment_location_property('dlPropB', 'B DL Prop')
+        self.assertEqual(builder.result, {
+            'propA': 'A Prop',
+            'systemProperties': {
+                'sysA': 'A',
+                'sysB': 'B'
+            }, 
+            'deploymentLocationInst': {
+                'name': 'Test',
+                'type': 'Kubernetes',
+                'properties': {
+                    'dlPropA': 'A DL Prop',
+                    'dlPropB': 'B DL Prop'
+                }
+            }
+        })
+
+    def test_deployment_location_property_not_hidden(self):
+        system_properties, properties, deployment_location = self.__example_values()
+        properties['deploymentLocation'] = 'Test'
+        builder = ResourceContextBuilder(system_properties, properties, deployment_location)
+        self.assertEqual(builder.result, {
+            'propA': 'A Prop',
+            'deploymentLocation': 'Test',
+            'systemProperties': {
+                'sysA': 'A',
+                'sysB': 'B'
+            }, 
+            'deploymentLocationInst': {
+                'name': 'Test',
+                'type': 'Kubernetes',
+                'properties': {
+                    'dlPropA': 'A DL Prop'
+                }
+            }
+        })
```

Comment:
Can't see any tests where you try to set property or properties with key "systemProperties" and "deploymentLocationInst" (which are covered in the code) - it's possible I've missed these tests though!

Reply:
you're right, they are missing. I will add them.